geth/api/api_test.go

```diff
@@ -0,0 +1,125 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/status-im/status-go/geth/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateNodeConfig(t *testing.T) {
+	noErrorsCallback := func(resp common.APIDetailedResponse) {
+		require.True(t, resp.Status, "expected status equal true")
+		require.Empty(t, resp.FieldErrors)
+		require.Empty(t, resp.Message)
+	}
+
+	testCases := []struct {
+		Name     string
+		Config   string
+		Callback func(common.APIDetailedResponse)
+	}{
+		{
+			Name: "response for valid config",
+			Config: `{
+				"NetworkId": 1,
+				"DataDir": "/tmp"
+			}`,
+			Callback: noErrorsCallback,
+		},
+		{
+			Name:   "response for invalid JSON string",
+			Config: `{"Network": }`,
+			Callback: func(resp common.APIDetailedResponse) {
+				require.False(t, resp.Status)
+				require.Contains(t, resp.Message, "validation: invalid character '}'")
+			},
+		},
+		{
+			Name:   "response for config with multiple errors",
+			Config: `{}`,
+			Callback: func(resp common.APIDetailedResponse) {
+				required := map[string]string{
+					"NodeConfig.NetworkID": "required",
+					"NodeConfig.DataDir":   "required",
+				}
+
+				require.False(t, resp.Status)
+				require.Contains(t, resp.Message, "validation: validation failed")
+				require.Equal(t, 2, len(resp.FieldErrors))
+
+				for _, err := range resp.FieldErrors {
+					require.Contains(t, required, err.Parameter)
+					require.Contains(t, err.Error(), required[err.Parameter])
+				}
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Logf("TestValidateNodeConfig: %s", tc.Name)
+		testValidateNodeConfig(tc.Config, tc.Callback)
+	}
+}
+
+func testValidateNodeConfig(config string, fn func(common.APIDetailedResponse)) {
```

Comment:
This function is used once. It's better to inline this in `TestValidateNodeConfig`. Just initialize `statusAPI := StatusAPI{}` before loop.

```go
api := NewStatusAPI()
for _, tc := range testCases {
  resp := api.ValidateJSONConfig(tc.Config)
  tc.Callback(resp)
}
```